components/script/dom/bindings/buffer_source.rs

```diff
@@ -226,27 +216,138 @@ where
             BufferSource::Float64Array(buffer) |
             BufferSource::DataView(buffer) |
             BufferSource::ArrayBuffer(buffer) |
-            BufferSource::Default(buffer) => {
+            BufferSource::Default(buffer) => unsafe {
+                IsDetachedArrayBufferObject(*buffer.handle())
```

Comment:
Same thing here. 

Basically, for `ArrayBuffer` we can call `IsDetachedArrayBufferObject` directly, for all the others we should first root the buffer from the view, as done [here](https://searchfox.org/mozilla-central/rev/345ec3c55ddda5f0ce37168f0644dcdcc4834409/dom/streams/ReadableStreamBYOBReader.cpp#232), and then call `IsDetachedArrayBufferObject` on the rooted buffer([like here](https://searchfox.org/mozilla-central/rev/345ec3c55ddda5f0ce37168f0644dcdcc4834409/dom/streams/ReadableStreamBYOBReader.cpp#246)).